### PR TITLE
Fixing RingaEvent so that it can handle being caught by multiple cont…

### PR DESCRIPTION
**Pull request by joshjung**
…rollers

### CODE REVIEW

File: src/RingaEvent.js [hunk 1]
```diff
@@ -137,6 +142,16 @@ class RingaEvent extends RingaObject {
...
+    this.catchers.push(controller);
+  }
+
+  _uncatch(controller) {
+    let ix = this.catchers.indexOf(controller);
+    this.catchers.splice(ix, 1);
+  }
+
   /**
    * Completely kills the current Ringa thread, keeping any subsequent executors from running.
```
Comment (corragon) [inline src/RingaEvent.js]: This comment doesn't appear to match the function anymore

File: src/RingaEvent.js [hunk 2]
```diff
@@ -154,12 +169,25 @@ class RingaEvent extends RingaObject {
     //return `RingaEvent [ '${this.type}' caught by ${this.controller ? this.controller.toString() : ''} ] `;
```
Comment (corragon) [inline src/RingaEvent.js]: What?  Why is this here?  toString() probably shouldn't return 'whatever'

File: src/RingaEvent.js [hunk 3]
```diff
@@ -154,12 +169,25 @@ class RingaEvent extends RingaObject {
...
+      } else {
+        this._dispatchEvent(RingaEvent.DONE);
+      }
+    }
+
   }
 
-  _fail(error) {
-    this._dispatchEvent(RingaEvent.FAIL, undefined, error);
+  _fail(controller, error) {
```
Comment (corragon) [inline src/RingaEvent.js]: Expected to see ` this.pushError(error);` in here, not in `fail()`

File: src/RingaEvent.js (hunk 1 above)
Comment (corragon) [inline src/RingaEvent.js]: This comment doesn't appear to match the function anymore

File: src/RingaEvent.js (hunk 3 above)
Comment (joshjung) [inline src/RingaEvent.js]: That will be a bug we fix later
